torch/_inductor/codegen/cpp.py

```diff
@@ -2606,6 +2607,18 @@ def select_tiling_indices():
                         non_contig_stride_const.add(int(var.name[1:]))
                     else:
                         non_contig_stride_other.add(int(var.name[1:]))
+
+            # If there has index_expr, remove this var in index_expr from contig_vars
```

Comment:
I don't understand why we are removing it here...

Reply:
Yes, the original fix seems a patch fixing, I update the code.